examples/dispatch_call_on_tick/main_test.go

```diff
@@ -18,34 +19,56 @@ import (
 )
 
 func TestPluginContext_OnTick(t *testing.T) {
-	opt := proxytest.NewEmulatorOption().WithVMContext(&vmContext{})
-	host, reset := proxytest.NewHostEmulator(opt)
-	defer reset()
-
-	// Call OnVMStart.
-	require.Equal(t, types.OnPluginStartStatusOK, host.StartPlugin())
-	require.Equal(t, tickMilliseconds, host.GetTickPeriod())
-
-	for i := 1; i < 10; i++ {
-		host.Tick() // call OnTick
-		attrs := host.GetCalloutAttributesFromContext(proxytest.PluginContextID)
-		// Verify DispatchHttpCall is called
-		require.Equal(t, len(attrs), i)
-		// Receive callout response.
-		host.CallOnHttpCallResponse(attrs[0].CalloutID, nil, nil, nil)
-		// Check Envoy logs.
-		logs := host.GetInfoLogs()
-		require.Contains(t, logs, fmt.Sprintf("called %d for contextID=%d", i, proxytest.PluginContextID))
-	}
+	vmTest(t, func(t *testing.T, vm types.VMContext) {
+		opt := proxytest.NewEmulatorOption().WithVMContext(vm)
+		host, reset := proxytest.NewHostEmulator(opt)
+		defer reset()
 
+		// Call OnVMStart.
+		require.Equal(t, types.OnPluginStartStatusOK, host.StartPlugin())
+		require.Equal(t, tickMilliseconds, host.GetTickPeriod())
+
+		for i := 1; i < 10; i++ {
+			host.Tick() // call OnTick
+			attrs := host.GetCalloutAttributesFromContext(proxytest.PluginContextID)
+			// Verify DispatchHttpCall is called
+			require.Equal(t, len(attrs), i)
+			// Receive callout response.
+			host.CallOnHttpCallResponse(attrs[0].CalloutID, nil, nil, nil)
+			// Check Envoy logs.
+			logs := host.GetInfoLogs()
+			require.Contains(t, logs, fmt.Sprintf("called %d for contextID=%d", i, proxytest.PluginContextID))
+		}
+	})
 }
 
 func TestPluginContext_OnVMStart(t *testing.T) {
-	opt := proxytest.NewEmulatorOption().WithVMContext(&vmContext{})
-	host, reset := proxytest.NewHostEmulator(opt)
-	defer reset()
+	vmTest(t, func(t *testing.T, vm types.VMContext) {
+		opt := proxytest.NewEmulatorOption().WithVMContext(vm)
+		host, reset := proxytest.NewHostEmulator(opt)
+		defer reset()
+
+		// Call OnVMStart.
+		require.Equal(t, types.OnPluginStartStatusOK, host.StartPlugin())
+		require.Equal(t, tickMilliseconds, host.GetTickPeriod())
+	})
+}
+
+func vmTest(t *testing.T, f func(*testing.T, types.VMContext)) {
```

Comment:
Please add the detailed comment what this does and what's each subtests doing

Reply:
Added doc